global/utils/keycloakUtils.ts

```diff
@@ -21,3 +28,38 @@ export const scopesFromPermissions = (permissions: Permission[]) => {
 		.filter(({ scopes }) => scopes)
 		.flatMap(({ rsname, scopes }) => scopes.flatMap((scope) => [rsname + '.' + scope]));
 };
+
+export const refreshAccessToken = async (refreshToken: string) => {
+	try {
+		const url = urlJoin(
+			NEXT_PUBLIC_KEYCLOAK_HOST,
+			`realms`,
+			NEXT_PUBLIC_KEYCLOAK_REALM,
+			'protocol/openid-connect/token',
+		);
+
+		const formData = new URLSearchParams({
+			client_id: NEXT_PUBLIC_KEYCLOAK_CLIENT_ID,
+			client_secret: KEYCLOAK_CLIENT_SECRET,
+			grant_type: 'refresh_token',
+			refresh_token: refreshToken,
+		});
+
+		const response = await fetch(url, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+			},
+			body: formData,
+		});
+
+		// Parse the response and return the result
+		if (response.ok) {
+			return await response.json();
+		}
+		return;
+	} catch (error) {
+		console.error('Error during token refresh:', error);
+		return;
```

Comment:
those `return`s are implicit, no need to put them here
```suggestion
	} catch (error) {
		console.error('Error during token refresh:', error);
```

alternatively, if you wanted to return something beyond `undefined` that would cover both if/else cases, a single one after line 64 would do the trick without repeating code